File: notifier-client/src/utils/models/Reminder.ts
export interface Reminder {
  reminderId: number | null;
  userId: number;
  title: string;
  notes: string;
  expiration: Date;
  repeated: boolean;
  repeatPattern: string;
  completed: boolean;
  deleted: boolean;
  createdAt: Date | null;
  updatedAt: Date | null;
}

export interface CreateReminderResult {
  success: boolean;
  data?: Reminder;
  error?: string;
}

export interface GetRemindersResult {
  success: boolean;
  data?: Reminder[];
  error?: string;
}

export interface UpdateReminderResult {
  success: boolean;
  data?: Reminder;
  error?: string;
}

export interface ReminderApiResponse {
  reminder_id: number | null;
  user_id: number;
  title: string;
  notes: string;
  expiration: Date;
  repeated: boolean;
  repeat_pattern: string;
  completed: boolean;
  deleted: boolean;
  created_at: Date | null;
  updated_at: Date | null;
}

export const transformReminderFromApi = (data: ReminderApiResponse): Reminder => ({
  reminderId: data.reminder_id,
  userId: data.user_id,
  title: data.title,
  notes: data.notes,
  expiration: data.expiration,
  repeated: data.repeated,
  repeatPattern: data.repeat_pattern,
  completed: data.completed,
  deleted: data.deleted,
  createdAt: data.created_at,
  updatedAt: data.updated_at,
});

File: notifier-server/models/reminder.go
package models

import (
	"time"
)

type Reminder struct {
    ReminderID    	int       	`json:"reminder_id"`
    UserID    		int       	`json:"user_id"`
    Title     		string    	`json:"title"`
    Notes  			string    	`json:"notes"`
	Expiration      time.Time 	`json:"expiration"`
	Repeated		bool	  	`json:"repeated"`
	RepeatPattern   string    	`json:"repeat_pattern"`
	Completed		bool      	`json:"completed"`
    Deleted   		bool      	`json:"deleted"`
    CreatedAt 		time.Time 	`json:"created_at"`
    UpdatedAt 		*time.Time 	`json:"updated_at"`
    UserIdFromAPI int     		`json:"userid"`
}